fastlane/lib/fastlane/actions/xcode_select.rb

```diff
@@ -18,32 +18,92 @@ module Actions
     #   environment variable into the full Developer content path.
     #
     class XcodeSelectAction < Action
-      def self.run(params)
-        params = nil unless params.kind_of? Array
-        xcode_path = (params || []).first
 
-        # Verify that a param was passed in
-        UI.user_error!("Path to Xcode application required (e.x. \"/Applications/Xcode.app\")") unless xcode_path.to_s.length > 0
+      def self.find_xcode(req)
+        req = Gem::Requirement.new(req.to_s)
+
+        Gem::Specification.find_by_name('xcode-install')
+        require 'xcode/install'
+
+        installer = XcodeInstall::Installer.new
+        installed = installer.installed_versions.reverse
+        installed.detect do |xcode|
+          req.satisfied_by? Gem::Version.new(xcode.version)
+        end
+      rescue Gem::LoadError
+        UI.error("The 'xcode-install' gem is needed to select Xcode by version number.")
+        Actions.verify_gem!('xcode-install') # This is awkward and destined to fail but is used to get the install help messages
+      end
+
+      def self.run(params)
+        xcode_path =
+          case
+          when params[:path]
+            params[:path]
+          when version = params[:version]
+            xcode = find_xcode(version)
+            UI.user_error!("Cannot find an installed Xcode satisfying '#{version}'") if xcode.nil?
 
-        # Verify that a path to a directory was passed in
-        UI.user_error!("Path '#{xcode_path}' doesn't exist") unless Dir.exist?(xcode_path)
+            UI.verbose("Found Xcode version #{xcode.version} at #{xcode.path} satisfying requirement #{version}")
+            xcode.path
+          else
+            UI.user_error!("path or version must be specified")
+          end
 
         UI.message("Setting Xcode version to #{xcode_path} for all build steps")
 
         ENV["DEVELOPER_DIR"] = File.join(xcode_path, "/Contents/Developer")
       end
 
       def self.description
-        "Change the xcode-path to use. Useful for beta versions of Xcode"
+        "Select which Xcode to use, either by path or version"
       end
 
       def self.author
         "dtrenz"
       end
 
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(key: :path,
```

Comment:
@KrauseFx This is the `xcode_select` action. I had to introduce a `path` keyword to support that behaviour. This is the part I'm most uncomfortable with because it breaks backwards compatibility with the existing version. That is:

```ruby
xcode_select("/Applications/Xcode.app")
```

would have to be updated to be:

```ruby
xcode_select(path: "/Applications/Xcode.app")
```